static/app/components/charts/miniBarChart.tsx

```diff
@@ -273,12 +314,26 @@ function MiniBarChart({
         updated.stack = 'stack1';
       }
       set(updated, 'itemStyle.color', colorList[i]);
-      set(updated, 'itemStyle.borderRadius', [1, 1, 0, 0]); // Rounded corners on top of the bar
       set(updated, 'emphasis.itemStyle.color', emphasisColors?.[i] ?? colorList[i]);
+
+      if (stacked) {
+        set(updated, 'data', updateDataItemBorderRadius(groupedData, original.data, i));
```

Comment:
```suggestion
      if (stacked) {
        const groupedData = groupDataByName(series ?? []);
        set(updated, 'data', updateDataItemBorderRadius(groupedData, original.data, i));
```

Reply:
this is inside a loop, and we don't want it to occur every time. it is better if we have `groupDataByName` outside